scim2-sdk-common/src/main/java/com/unboundid/scim2/common/filters/ComparisonFilter.java

```diff
@@ -22,7 +22,14 @@
 import com.unboundid.scim2.common.utils.JsonUtils;
 
 /**
- * Attribute comparison filter.
+ * This superclass represents filter types that are used to compare attribute
+ * values of SCIM resources. For example, an {@link EqualFilter} is a comparison
```

Comment:
Nit: Put the example in a separate paragraph. 

My thinking here is that `ComplexValueFilter` establishes a good, readable pattern that would also work well here:

1. Summary/definition
2. Example
3. Supplementary information

This comment — if you choose to accept it — also applies to other classes, such as `ContainsFilter` and `EndsWithFilter`.